### ISSUE TITLE
Race condition in test_xss_scanner

### ISSUE THREAD

**Issue by eli-schwartz** [mitmproxy/mitmproxy#3275]
##### Steps to reproduce the problem:

```
git clone https://github.com/mitmproxy/mitmproxy
cd mitmproxy
python setup.py pytest
```
##### Result:

```
====================================================================== FAILURES ======================================================================
______________________________________________________ TestXSSScanner.test_find_unclaimed_URLs _______________________________________________________

self = <test.examples.test_xss_scanner.TestXSSScanner object at 0x7f091866cb38>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f091866cdd8>
logger = <test.examples.test_xss_scanner.TestXSSScanner.logger.<locals>.Logger object at 0x7f091866ce10>

    def test_find_unclaimed_URLs(self, monkeypatch, logger):
        logger.args = []
        monkeypatch.setattr("mitmproxy.ctx.log", logger)
        monkeypatch.setattr("socket.gethostbyname", self.mocked_socket_gethostbyname)
        xss.find_unclaimed_URLs("<html><script src=\"http://google.com\"></script></html>",
                                "https://example.com")
        assert logger.args == []
        xss.find_unclaimed_URLs("<html><script src=\"http://unclaimedDomainName.com\"></script></html>",
                                "https://example.com")
>       assert logger.args[0] == 'XSS found in https://example.com due to unclaimed URL "http://unclaimedDomainName.com".'
E       IndexError: list index out of range

test/examples/test_xss_scanner.py:329: IndexError
================================================================== warnings summary ==================================================================
test/mitmproxy/utils/test_debug.py::test_dump_info
  /build/mitmproxy/src/mitmproxy/mitmproxy/utils/debug.py:70: RuntimeWarning: coroutine 'Script.watcher' was never awaited
    gc.collect()

-- Docs: http://doc.pytest.org/en/latest/warnings.html
================================================= 1 failed, 1416 passed, 1 warnings in 86.32 seconds =================================================
```

**Comment by eli-schwartz** [comment]
The tests pass okay, if I use a faster/more powerful computer to run them. e.g. they fail on my laptop, but pass on this build server: https://pkgbuild.com/

**Comment by elijahbal** [comment]
I can confirm that it is consistently failing on my i3 local computer.

```python

============================================================= FAILURES =============================================================
_____________________________________________ TestXSSScanner.test_find_unclaimed_URLs ______________________________________________

self = <test.examples.test_xss_scanner.TestXSSScanner object at 0x7f7f8e9ddac8>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f7f8e9dda20>
logger = <test.examples.test_xss_scanner.TestXSSScanner.logger.<locals>.Logger object at 0x7f7f8e9dd470>

    def test_find_unclaimed_URLs(self, monkeypatch, logger):
        logger.args = []
        monkeypatch.setattr("mitmproxy.ctx.log", logger)
        monkeypatch.setattr("socket.gethostbyname", self.mocked_socket_gethostbyname)
        xss.find_unclaimed_URLs("<html><script src=\"http://google.com\"></script></html>",
                                "https://example.com")
        assert logger.args == []
        xss.find_unclaimed_URLs("<html><script src=\"http://unclaimedDomainName.com\"></script></html>",
                                "https://example.com")
>       assert logger.args[0] == 'XSS found in https://example.com due to unclaimed URL "http://unclaimedDomainName.com".'
E       IndexError: list index out of range

test/examples/test_xss_scanner.py:329: IndexError
============================================ 1 failed, 1422 deselected in 2.99 seconds =============================================
```

**Comment by elijahbal** [comment]
#3294 should fix it

**Comment by elijahbal** [comment]
This is fixed in #3294 #
Issue should be close

**Comment by mhils** [comment]
Thanks for pointing this out!
